backend/app/summary.py

```diff
@@ -253,6 +251,39 @@ def update(sqldb) -> None:
         }
     )
 
+    # collect all app IDs to update
+    apps_to_update = {}
+    for app_id, data in summary_dict.items():
+        try:
+            summary_json = json.loads(json.dumps(data, cls=JSONSetEncoder))
+            apps_to_update[app_id] = summary_json
+        except Exception as e:
+            print(f"Error encoding summary data for {app_id}: {str(e)}")
+            continue
+
+    # update all apps in a single transaction
+    try:
+        for app_id, summary_json in apps_to_update.items():
+            app = models.App.by_appid(sqldb, app_id)
+            if app:
+                appstream_data = app.appstream
+                app.summary = summary_json
+                if appstream_data:
+                    app.appstream = appstream_data
+                sqldb.session.add(app)
+            else:
+                app = models.App(
+                    app_id=app_id,
+                    type="desktop-application",
```

Comment:
Could this cause problems/be wrong?

`summary.update()` also always runs after `load_appstream`, so it could be fine, to not have the else at all here?

Reply:
Hopefully the type would be corrected by the next run of whatever, but it might be wrong on creation. Not sure, if the other logic will correct type. Would need to check.